modules/apps/blogs/blogs-web/src/main/java/com/liferay/blogs/web/internal/display/context/BlogsEditEntryDisplayContext.java

```diff
@@ -566,7 +567,14 @@ private long[] _getAvailableFriendlyURLAssetCategoryIds() {
 			return new long[0];
 		}
 
-		_assetCategoryIds = assetEntry.getCategoryIds();
+		List<Long> assetCategoryIdsList = ListUtil.fromArray(
+			assetEntry.getCategoryIds());
+
+		assetCategoryIdsList.removeAll(
+			ListUtil.fromArray(_getCurrentFriendlyURLAssetCategoryIds()));
+
+		_assetCategoryIds = ArrayUtil.toArray(
+			assetCategoryIdsList.toArray(new Long[0]));
```

Comment:
Why are you mixing `Long[]` and `long[]` here? These are two distinct, non-compatible types.

You should use `ArrayUtil.toLongArray` instead.

Reply:
@adolfopa and @ambrinchaudhary 

Do you think we could move this forward?